Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1605,15 +1617,31 @@ namespace Babylon
             XRRigidTransform& m_transform;
             Napi::ObjectReference m_jsPose{};
             Napi::ObjectReference m_jsJointPose{};
+            
+            bool m_hasBegunTracking = false;
 
             Napi::Value GetViewerPose(const Napi::CallbackInfo& info)
             {
+                // To match the WebXR implementation we should return undefined here until we have gotten
+                // initial tracking. After that point we can just continue returning the position
+                // as it will be marked as estimated if tracking is lost.
+                if (!m_hasBegunTracking && !m_frame->IsTracking())
+                {
+                    return info.Env().Undefined();
```

Comment:
```suggestion
                    return info.Env().Null();
```

Reply:
why is the preference null compared to undefined?

---
In reply to: [527103137](https://github.com/BabylonJS/BabylonNative/pull/504#discussion_r527103137) [](ancestors = 527103137)

Reply:
This completely depends on the contract of GetViewerPose. `null` and `undefined` are two different things in js. I'm assuming the contract is nullable?

Reply:
Actually, no, it's not. The contract is undefined or the pose.

https://github.com/BabylonJS/Babylon.js/blob/b8209e7d7c27abcf021d55a0cfc69aa6e0951a0d/src/LibDeclarations/webxr.d.ts#L130

Reply:
Keeping it as undefined. Thanks for finding the source of truth on that one @bghgary

Reply:
Interesting. The docs I was looking at before indicated null, but I guess we should go by the d.ts.